docs/reference/index.rst

```diff
@@ -0,0 +1,44 @@
+API Reference
+=============
+
+ImageIO's API follows the usual idea of choosing sensible defaults for the average use, but giving
+you fine grained control 
+
+Core API (Basic Usage)
+^^^^^^^^^^^^^^^^^^^^^^
+
+The API documented in this section is shared by all plugins
+
+.. toctree::
+    :maxdepth: 1
+
+    Core API v2.9 <userapi>
+
+
+Plugins & Backends (Advanced Usage)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sometimes, you need to do more than just "load an image, done". Sometimes you
+need to use a very specific feature of a very specific backend. ImageIOs allows
+you to do so by allowing its plugins to extend the core API. Typically this is
+done in the form of additional keyword arguments (``kwarg``). Below you can find
+a list of available plugins and which arguments they support.
```

Comment:
I think the supported formats page already fulfills this purpose to some extend. I don't seem to be able to get to the plugin-specific docs from the docs here. Also, plugins are classes, and the modules documented here seen to contain functions and classes aimed for internal use.

Reply:
Yes. The plugins & backends are still work in progress :) I haven't had time to finish this part yet, since I was working on a research paper deadline and was a bit pressed for time, but I am planning on working more on this during this week.

---

My idea was to refactor the supported formats page since it currently fulfills a double purpose / has a bit of a split personality. It is a big list of all the format strings that exist and, once you know the correct string, you can navigate to the backend-specific docs to tell you the kwargs. 

The title hints at "here are all the formats you can use", which it does list, but if my desire is to "figure out which plugin I should use to read my _image.XYZ_ file" then I would indeed check the "Supported Formats", but the current list is a bit confusing. I would have expected to find a list or table that I can search (ctrl+F) for ".XYZ" and then it lists all the format strings (or backends) that I could use here.

Similarly, if I know that I want/need to use backend X (e.g. imageio-ffmpeg) then I need to _know_ that I need to search for this information in "Supported Formats" (and click on the correct format string in the list). Since this is something API-related, I would, however, have searched for this in the API reference.

Long story short, I would like to change "Supported Formats" into lists of the form

```
- <extension> [Format Full Name](link to description/wikipedia/standard)
    Backend Libraries: <list of format strings or backends>

- PNG [Portable Network Graphic](https://en.wikipedia.org/wiki/Portable_Network_Graphics).
    Format String: :ref:`PNG-PIL <imageio.plugins.pillow>`, :ref:`PNG-FI <imageio.plugins.freeimage>`, ...
```

One list for _backends by format_ and one for _formats by backend_, or as a table (if I manage to figure out how to make a nice table in sphinx) so that we can maintain a single structure but index it by row or column for either of the two scenarios.

I would like to have this section here (Plugins & Backends) become the backend-specific docs, i.e., kwargs, recommended/supported versions, intricacies/gotchas, ...

Then again, I might be overthinking this and it is cleaner and easier to leave it as is?

Reply:
I think you're right. People can come at this from two directions and we should probably have two doc pages for that.

One note is that certain plugin-modules provide multiple plugin classes (e.g. FreeImage and Pil have a few specific classes for png and jpg iirc. Not sure if we'd want to list modules and then when you click one you get all the plugins defined in that module, or list all plugin classes to begin with (might be a bit much).

Reply:
@almarklein I noticed this while I was moving docstrings around. At the moment I have merged them all and indicated for which format a specific argument is available. Except for pillow, this seems to work with how things were documented. For pillow I'm not yet sure if I want to leave it as is, or if I want to refactor further.

In either case, this is something I will worry about after I refactored the supported formats. (we can still discuss the best approach though)